docs/csharp/language-reference/operators/null-coalescing-operator.md

```diff
@@ -1,31 +1,48 @@
 ---
-title: "?? operator - C# reference"
+title: "?? and ??= operators - C# reference"
 ms.custom: seodec18
-ms.date: 06/07/2019
+ms.date: 09/10/2019
 f1_keywords: 
   - "??_CSharpKeyword"
+  - "??=_CSharpKeyword"
 helpviewer_keywords: 
   - "null-coalescing operator [C#]"
   - "?? operator [C#]"
+  - "null-coalescing assignment [C#]"
+  - "??= operator [C#]"
 ms.assetid: 088b1f0d-c1af-4fe1-b4b8-196fd5ea9132
 ---
-# ?? operator (C# reference)
+# ?? and ??= operators (C# reference)
 
 The null-coalescing operator `??` returns the value of its left-hand operand if it isn't `null`; otherwise, it evaluates the right-hand operand and returns its result. The `??` operator doesn't evaluate its right-hand operand if the left-hand operand evaluates to non-null.
 
-The null-coalescing operator is right-associative, that is, an expression of the form
+Available in C# 8.0 and later, the null-coalescing assignment operator `??=` assigns the value of its right-hand operand to its left-hand operand only if the left-hand operand evaluates to `null`. The `??=` operator doesn't evaluate its right-hand operand if the left-hand operand evaluates to non-null.
+
+[!code-csharp[null-coalescing assignment](~/samples/csharp/language-reference/operators/NullCoalescingOperator.cs#Assignment)]
+
+The left-hand operand of the `??=` operator must be a variable, a [property](../../programming-guide/classes-and-structs/properties.md), or an [indexer](../../programming-guide/indexers/index.md) element. For more information about null-coalescing assignment, see the [feature proposal note](~/_csharplang/proposals/csharp-8.0/null-coalescing-assignment.md).
+
+In C# 7.3 and earlier, the type of the left-hand operand of the `??` operator must be either a reference type or a [nullable value type](../../programming-guide/nullable-types/index.md). Beginning with C# 8.0, that requirement is replaced with the following: the type of the left-hand operand of the `??` and `??=` operators cannot be a non-nullable value type. In particular, you can use the null-coalescing operators with unconstrained type parameters in C# 8.0 and later:
+
+[!code-csharp[unconstrained type parameter](~/samples/csharp/language-reference/operators/NullCoalescingOperator.cs#UnconstrainedType)]
+
+The null-coalescing operators are right-associative. That is, expressions of the form
 
 ```csharp
 a ?? b ?? c
+d ??= e ??= f
 ```
 
-is evaluated as
+are evaluated as
 
 ```csharp
 a ?? (b ?? c)
+d ??= (e ??= f)
 ```
 
-The `??` operator can be useful in the following scenarios:
+## Use cases
```

Comment:
nit (for consistency with other articles)

```suggestion
## Examples
```

Reply:
Applied, thanks @BillWagner